sdk/test-utils/recorder/test/browser/recorder.spec.ts

```diff
@@ -0,0 +1,296 @@
+import { RecorderEnvironmentSetup } from "../../src/utils";
+import chai from "chai";
+import { record } from "../../src";
+import xhrMock from "xhr-mock";
+const { expect } = chai;
+
+const expectedHttpResponse = "Hello World!";
+
+async function helloWorldRequest(): Promise<string> {
+  return new Promise((resolve) => {
+    function reqListener(this: any) {
+      resolve(this.responseText);
+    }
+
+    const req = new XMLHttpRequest();
+    req.addEventListener("load", reqListener);
+    req.open("GET", (window as any).__env__.PATH!);
+    req.send();
+  });
+}
+
+const recorderEnvSetup: RecorderEnvironmentSetup = {
+  replaceableVariables: {
+    PATH: "/replaced"
+  },
+  customizationsOnRecordings: [],
+  queryParametersToSkip: []
+};
+
+/**
+ * A function that generates another function with a predictable shape, even after compiling it to the browser.
+ */
+const getNoOpFunction = () => eval("() => {}");
```

Comment:
Why do we need to use eval here?

Reply:
So, the browser output, even of an empty function like this, can contain references to indexes of loops of the environment that is generated. These references have unique elements, pretty much unique each time the files change. It's an acceptable behavior usually, but during tests it makes it really hard to have a predictable behavior. Without this, the hashes in this tests will change even by adding a new line on the file.

Reply:
maybe try using the function keyword instead of arrow so you don't have to worry about `this` binding?

Reply:
Fixed! Thank you for the feedback!